modules: trusted-firmware-m: fix dependencies for isolation level

ioannisg opened the pull request:
Do not allow the TFM_ISOLATION_LEVEL to be non-hidden option,
if the IPC (PSA_API) option is not set, since, in that case,
only Isolation Level 1 is supported.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

ioannisg committed:
modules: trusted-firmware-m: fix dependencies for isolation level

Do not allow the TFM_ISOLATION_LEVEL to be non-hidden option,
if the TFM_IPC (PSA_API) option is not set, since, in that case,
only Isolation Level 1 is supported.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

ioannisg commented:
This will probably fail, now, since stm32 l552ze_2 forces TFM isolation level to 2, while it still allows the psa_level1 sample to build and run.

oyvindronningstad commented on modules/trusted-firmware-m/Kconfig:
```diff
@@ -120,17 +120,19 @@ config TFM_CMAKE_BUILD_TYPE_DEBUG
 	  1,2 or 3; the default is set by build configuration. When TF-M
 	  Profile option is supplied, do not allow manual setting of the
 	  isolation level, as it is determined by the profile setting.
+	  The isolation level is to be fixed to 1 if TF-M is not build with
+	  PSA_API (IPC) support.
```
```suggestion
	  Isolation levels 2 and 3 require PSA_API (IPC) support.
```

erwango replied:
```suggestion
	  As isolation levels 2 and 3 require PSA_API (IPC) support, force level 1 when not enabled.
```

ioannisg commented:
@erwango @oyvindronningstad thanks for the review - text updated.

ioannisg committed:
boards: do not force TF-M isolation level 2 for STM32L562/L552 boards

We do not need to force TFM Isolation Level 2 for the
STM32 board that can build with TM-M. We remove the
respective Kconfig setting. We also remove the Kconfig
overlay for STM32L562 board in the psa_level1 sample,
which was used for forcing Isolation Level 1; this is
not needed any more.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

ioannisg commented:
I added a commit that removes the forcing of TF-M isolation level to 2 for STM32 boards that support TFM - there does not seem to be any need for this, and the proof is the fact that psa_level1 sample could run successfully for the STM32L562 board (there was a Kconfig overlay that was explicitly setting the isolation level to 1)

ioannisg commented:
Will test psa_level1 on nucleo_l552ze_q before merging it.

ioannisg committed:
samples: psa_level1: add nucleo_l552ze_q_ns to allowed platforms

Add nucleo_l552ze_q_ns boards to the list of platforms
that can build and run the psa_level1 tf-m integration
sample. We add a dts overlay as well.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

ioannisg commented:
Need an overlay for the nucleo board. Wil add a re-run CI.

ioannisg commented:
Tested the integration samples  on nucleo_l552ze_q_ns
